### PR TITLE
[#2968] `prevent-body-scroll` prop for `va-modal` to prevent document scroll

### PR DESCRIPTION
**Pull request by aluarius**
Closes: #2968

It really doesn't exist for the current VaModal. For QA empty project is needed.

## Description
- [x] `prevent-body-scroll` prop that set the body position to fixed;
- [x] tested for spa and ssr;
- [x] minor docs updates. 

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: packages/docs/src/locales/en/en.json [hunk 1]
```diff
@@ -332,6 +332,7 @@
         "overlayOpacity": "Set the overlay's opacity",
         "anchorClass": "Set class name to the `anchor` slot container",
         "zIndex": "Set the modal's `z-index`",
+        "fixBody": "Set `body` position to `fixed` to prevent document scroll while modal is open.",
```
Comment (rustem-nasyrov) [inline packages/docs/src/locales/en/en.json:335]: Imo, poor naming. I understand that this property adds `position: fixed`, but it looks like we fixing something in the `document.body`. I think there will fit `preventScroll` or something similar.

File: packages/ui/src/components/va-modal/VaModal.vue [hunk 2]
```diff
@@ -295,14 +297,25 @@ export default defineComponent({
...
     useBlur(toRef(props, 'blur'), valueComputed)
 
+    const documentRef = useDocument()
+    const setBodyPosition = (position: string) => {
+      if (!documentRef.value || !props.fixBody) { return }
+
+      documentRef.value.body.style.position = position
+    }
+    const setFixedBodyPosition = () => setBodyPosition('fixed')
+    const resetBodyPosition = () => setBodyPosition('')
```
Comment (rustem-nasyrov) [inline packages/ui/src/components/va-modal/VaModal.vue:307]:
```suggestion
    const setBodyPosition = (position = '') => {
      if (!documentRef.value || !props.fixBody) { return }

      documentRef.value.body.style.position = position
    }
```

File: packages/ui/src/components/va-modal/VaModal.vue [hunk 3]
```diff
@@ -295,14 +297,25 @@ export default defineComponent({
...
+
     watch(valueComputed, newValueComputed => { // watch for open/close modal
       if (newValueComputed) {
         registerModal()
+        setFixedBodyPosition()
```
Comment (rustem-nasyrov) [inline packages/ui/src/components/va-modal/VaModal.vue:312]:
```suggestion
        setBodyPosition('fixed')
```

File: packages/ui/src/components/va-modal/VaModal.vue [hunk 4]
```diff
@@ -295,14 +297,25 @@ export default defineComponent({
...
         return
       }
 
       if (isLowestLevelModal.value) {
         freeFocus()
+        resetBodyPosition()
```
Comment (rustem-nasyrov) [inline packages/ui/src/components/va-modal/VaModal.vue:318]:
```suggestion
        setBodyPosition()
```

File: packages/docs/src/locales/en/en.json (hunk 1 above)
Comment (m0ksem) [reply packages/docs/src/locales/en/en.json:335]: preventPageScroll sounds good.

File: packages/ui/src/components/va-modal/VaModal.vue (hunk 4 above)
Comment (m0ksem) [reply packages/ui/src/components/va-modal/VaModal.vue:318]:
```suggestion
        setBodyPosition('')
```

**Review by m0ksem — CHANGES_REQUESTED** [review]
fixBody props is evil

**Review by rustem-nasyrov — APPROVED** [review]

**Review by m0ksem — APPROVED** [review]

**Pull request merged**